kernel/nano_core/src/lib.rs

```diff
@@ -130,6 +131,16 @@ pub extern "C" fn nano_core_start(
     );
     println_raw!("nano_core_start(): booted via multiboot2 with boot info at {:#X}.", multiboot_information_virtual_address); 
 
+    let rsdp_address = boot_info
+        .rsdp_v2_tag()
+        .map(|tag| tag.signature())
+        .or_else(|| boot_info.rsdp_v1_tag().map(|tag| tag.signature()))
+        .and_then(|utf8_result| utf8_result.ok())
+        .map(|signature| signature as *const _ as *const () as usize)
+        .and_then(|rsdp_address| PhysicalAddress::new(rsdp_address));
+
+    println_raw!("nano_core_start(): acquired RSDP address: {:#?}", rsdp_address);
```

Comment:
tiny nit: pretty-print format specifier `#` splits `Option`s into 3 lines, no need for that.
```suggestion
    println_raw!("nano_core_start(): bootloader provided RSDP address: {:X?}", rsdp_address);
```